packages/apps/plugins/plugin-space/src/SpacePlugin.tsx

```diff
@@ -40,19 +66,39 @@ export const SpacePlugin = (): PluginDefinition<SpacePluginProvides> => {
     ready: async (plugins) => {
       const clientPlugin = findPlugin<ClientPluginProvides>(plugins, 'dxos.org/plugin/client');
       const treeViewPlugin = findPlugin<TreeViewPluginProvides>(plugins, 'dxos.org/plugin/treeview');
-      const graphPlugin = findPlugin<GraphPluginProvides>(plugins, 'dxos.org/plugin/graph');
+      const sessionPlugin = findPlugin<SessionPluginProvides>(plugins, 'dxos.org/plugin/session');
       if (!clientPlugin) {
         return;
       }
 
+      addNodes({
+        id: SPACE_PLUGIN,
+        label: ['plugin name', { ns: SPACE_PLUGIN }],
+      });
+
+      addRelations(
+        { by: SPACE_PLUGIN, of: NAVMENU_ROOT, as: 'navmenu-parent' },
+        { by: NAVMENU_ROOT, of: SPACE_PLUGIN, as: 'navmenu-child' },
+      );
+
       const client = clientPlugin.provides.client;
       subscriptions.add(
         client.spaces.subscribe((spaces) => {
           spaceSubs.clear();
           const spaceIndices = getIndices(spaces.length);
           spaces.forEach((space, index) => {
+            const spaceNode = spaceToSessionNode(space, spaceIndices[index]);
+            const { added } = addNodes(spaceNode);
+            Object.keys(added).length > 0 &&
```

Comment:
how are nodes/relations removed?

Reply:
how do plugins for markdown/kanban/etc. get called when space nodes are added/change?